test/syncedEnforcer.test.ts

```diff
@@ -0,0 +1,82 @@
+import { Enforcer, newEnforcer, newModel, newSyncedEnforcer } from '../src';
+
+test('TestGFunctionInSyncedEnforcerWithRoles', async () => {
+  const m = newModel();
+  m.addDef('r', 'r', 'sub, obj, act');
+  m.addDef('p', 'p', 'sub, obj, act');
+  m.addDef('g', 'g', '_, _');
+  m.addDef('e', 'e', 'some(where (p.eft == allow))');
+  m.addDef('m', 'm', 'g(r.sub, p.sub) && r.obj == p.obj && r.act == p.act');
+
+  const e = await newSyncedEnforcer(m);
+
+  await e.addPermissionForUser('alice', 'data1', 'read');
+  await e.addPermissionForUser('bob', 'data2', 'write');
+  await e.addPermissionForUser('data2_admin', 'data2', 'read');
+  await e.addPermissionForUser('data2_admin', 'data2', 'write');
+
+  // Synced enforcer should not fail to recognize subject
+  expect(e.enforceSync('alice', 'data1', 'read')).toBe(true);
```

Comment:
I don't know if we should also test that result matches with  `e.enforcer`.

Reply:
I don't think that is necessary.